client/my-sites/checkout/checkout-thank-you/hundred-year-thank-you/index.tsx

```diff
@@ -169,13 +182,20 @@ export default function HundredYearThankYou( {
 
 	if (
 		! isReceiptLoading &&
-		( ! receipt.data?.purchases?.length || receipt.data?.purchases[ 0 ].blogId !== siteId )
+		( ! receipt.data?.purchases?.length || receipt.data?.purchases[ 0 ].blogId !== siteId ) &&
+		// For transfers, the current siteId might be different - purchase performed with no site (siteId = null)
+		// and blog created after the purchase (siteId != null).
+		productSlug !== domainProductSlugs.TRANSFER_IN
 	) {
 		page( '/' );
 	}
 
 	const isMobile = useMobileBreakpoint();
-	const isPageLoading = isReceiptLoading || isLoadingDomains;
+	const isDomainDataLoaded = ! isLoadingDomains && targetDomain !== null;
+	const isPageLoading =
+		isReceiptLoading ||
+		isLoadingDomains ||
+		( productSlug !== PLAN_100_YEARS && ! isDomainDataLoaded );
```

Comment:
We only want to check if the domain data is being loaded in case we're not within the 100-year plan flow. 
The other flows (100-year domain, 100-year transfer) need the queried data to load CTAs and other component labels/descriptions, but the 100-year plan one doesn't - that's why the `productSlug !== PLAN_100_YEARS && ! isDomainDataLoaded` check is performed here.